### ISSUE TITLE
Calendar: its possible to break the filter ui

### ISSUE THREAD

**Issue by tvannahl** [tine20/tine20#6903]
Please take special care for the left side of the screenshot. There should have been an entry that has a (+) and a (-) button. But in some cases this entry can be lost and the user is left with (-) button that behaves like a (+) button and without a (-) button in that row.

![broken filter ui](https://user-images.githubusercontent.com/662132/44273175-f95b9180-a23e-11e8-858d-6a17ec779255.png)

Unfortunately I was since the initial view of this bug only able to reproduce the bug once and I am still not at a point where I can reliably reproduce the bug. But I thought I could report it anyway, maybe someone else is able to reproduce the bug in a reliable fashion.

**Comment by pschuele** [comment]
yep, this does indeed happen sometimes. I also have seen this before but could not reproduce it afterwards. it would be really great if someone could provide a valid reproducer for this.
